Issue #35 Curate the Live Channels Display on Homepage

CodeItQuick opened the pull request:
Changes made to show live channels sorted by view count, also displaying time online. Modified the Feeling Lucky button to display a random live channel.

CodeItQuick committed:
Issue #35 Curate the Live Channels Display on Hompage

Changes made to show live channels sorted by view count, also displaying time online. Modified the Feeling Lucky button to display a random live channel.

benrick commented on src/DevChatter.DevStreams.Infra.Dapper/DapperCrudRepository.cs:
```diff
@@ -86,6 +86,36 @@ public async Task<List<T>> GetAll<T>(string filter, string orderBy, object args)
+        /// Returns all the channel data for a channel object.
+        /// </summary>
+        /// <returns>All Channel Info in a List<Channel> Object </Channel></returns>
+        public async Task<List<Channel>> GetAllChannelInfo()
```
Check out the `IChannelAggregateService`, which should be able to get you the same information as this. You can use that instead of creating your own method.

CodeItQuick replied:
Had to create my own method, but it is in the new location, good? Or change again?

benrick commented on src/DevChatter.DevStreams.Core/Data/ICrudRepository.cs:
```diff
@@ -11,6 +11,7 @@ public interface ICrudRepository
         Task<List<T>> GetAll<T>();
         Task<List<T>> GetAll<T>(string filter, object args);
         Task<List<T>> GetAll<T>(string filter, string orderBy, object args);
+        Task<List<Channel>> GetAllChannelInfo();
```
`ICrudRepository` should be for generic methods and types only, so we won't want `Channel` to be known by it.

benrick commented on src/DevChatter.DevStreams.Core/Twitch/ChannelLiveState.cs:
```diff
@@ -1,8 +1,13 @@
     {
         public string TwitchId { get; set; }
         public bool IsLive { get; set; }
+        public DateTime startedAt { get; set; }
```
Use PascalCasing here.

benrick commented on src/DevChatter.DevStreams.Core/Twitch/ChannelLiveState.cs:
```diff
@@ -1,8 +1,13 @@
         public string TwitchId { get; set; }
         public bool IsLive { get; set; }
+        public DateTime startedAt { get; set; }
+        public int viewerCount { get; set; }
```
Use PascalCasing here.

benrick commented on src/DevChatter.DevStreams.Core/Twitch/TwitchResult.cs:
```diff
@@ -50,4 +50,30 @@ public class StreamResult
+    public class ChannelResultData
+    {
+        public int _total { get; set; }
+        public List<ChannelResult> follows { get; set; }
```
Use PascalCasing here.

benrick reviewed (changes requested):
Haven't fully reviewed, but general feedback would be to match the casing used elsewhere in the project. I mainly noticed the use of camelCasing instead of PascalCasing for properties.

Also, look at using the IChannelAggregateService instead of writing a new method to access Channel and Twitch info together. I think the aggregate does that already.

CodeItQuick committed:
Issue DevChatter#35 Curate the Live Channels Display on Hompage

Requested updates to the pull request. Moved the getall method to the channelaggregateservices where it should be. Also changed the properties to Camel Casing

CodeItQuick committed:
Issue #35 Curate the Live Channels Display on Homepage

Updated a missed camal case.

CodeItQuick committed:
Updated Feeling Lucky Button so that it works

Updated feeling lucky button

benrick commented on src/DevChatter.DevStreams.Core/Data/IChannelAggregateService.cs:
```diff
@@ -9,6 +9,7 @@ public interface IChannelAggregateService
         List<Channel> GetAll();
         List<Channel> GetAll(string userId);
         Channel GetAggregate(int id);
+        Task<List<Channel>> GetAllAggregate();
```
`GetAllAggregates`

benrick commented on src/DevChatter.DevStreams.Infra.Dapper/Services/ChannelAggregateService.cs:
```diff
@@ -92,6 +92,34 @@ public Channel GetAggregate(int id)
+                {
+                    if (twitchChannels.Where(x => x?.ChannelId == channel?.Id).Any())
+                    {
+                        channel.Twitch = twitchChannels.Where(x => x?.ChannelId == channel?.Id).First();
```
Get rid of the outer `if` and make the `Where` into a `SingleOrDefault` that includes the filtering lambda expression.

benrick commented on src/DevChatter.DevStreams.Infra.Twitch/TwitchStreamService.cs:
```diff
@@ -19,11 +20,20 @@ public TwitchStreamService(IOptions<TwitchSettings> twitchSettings)
-        /// Returns the subset of the channels which are currently live on Twitch.
-        /// </summary>
-        /// <param name="channelNames">Names of the Channels to check for live status.</param>
-        /// <returns>The names of the subset of channels that are currently live.</returns>
```
Why is this summary command gone from `GetChannelLiveStates`?

benrick committed:
Rearrange file to make diff simpler

benrick committed:
Remove redundant code

benrick committed:
non-functional code clean-up

benrick committed:
Rename GetAllAggregate to GetAllAggregates

benrick committed:
Remove unused TwitchAPI Result classes

benrick committed:
Simplify the TwitchChannel selection in GetAllAggregates

benrick committed:
Clean up spacing

benrick commented on src/DevChatter.DevStreams.Web/Controllers/IsLiveController.cs:
```diff
@@ -26,20 +27,47 @@ public class IsLiveController : Controller
+                .Select(x => x.ViewerCount)
+                .ToList();
+
+            var responseObject = new
```
Why aren't we just returning back a variation on the LiveState objects, which have the properties for Channel, ViewCount, TimeOnline?

benrick committed:
WIP - Refactoring the objects, linq, and time handling

benrick commented:
I want to get this merged soon, so I'll try to review it quickly on stream and get it merged in.

benrick committed:
Card Styling for Live Channels

benrick committed:
Merge branch 'dev' into pr/77
